test/extended/openstack/machinesets.go

```diff
@@ -0,0 +1,135 @@
+package openstack
+
+import (
+	"encoding/json"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	configv1 "github.com/openshift/api/config/v1"
+	machinev1 "github.com/openshift/api/machine/v1beta1"
+	framework "github.com/openshift/cluster-api-actuator-pkg/pkg/framework"
+	machineproviderv1 "github.com/openshift/machine-api-provider-openstack/pkg/apis/openstackproviderconfig/v1alpha1"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/networks"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/ports"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+var _ = g.Describe("[sig-installer][Suite:openshift/openstack] The OpenStack platform", func() {
+
+	defer g.GinkgoRecover()
+
+	var dc dynamic.Interface
+	var clientSet *kubernetes.Clientset
+
+	g.Context("after deletion of a machineset", func() {
+		g.BeforeEach(func() {
+			g.By("preparing openshift dynamic client")
+			cfg, err := e2e.LoadConfig()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			dc, err = dynamic.NewForConfig(cfg)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			clientSet, err = e2e.LoadClientset()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			g.By("preparing openstack client")
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+		})
+
+		g.It("should not have leftovers ports", func() {
+
+			skipUnlessMachineAPIOperator(dc, clientSet.CoreV1().Namespaces())
+			g.By("fetching worker machineSets")
+			var networkClient *gophercloud.ServiceClient
+			var rawBytes []byte
+			var t machineproviderv1.OpenstackProviderSpec
+			var rclient runtimeclient.Client
+
+			networkClient, err := client(serviceNetwork)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			machineSets, err := listWorkerMachineSets(dc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			if len(machineSets) == 0 {
+				e2eskipper.Skipf("Expects at least one worker machineset. Found none.")
+			}
+			xconfig, err := config.GetConfig()
```

Comment:
Is it needed to get the config again if it's already available at line 40 or it's a different one?

Reply:
I'm changing to e2e.LoadConfig , anyway cfg is not accessible here